examples/versioning-test.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"github.com/emicklei/go-restful"
+	"net/http"
+	"reflect"
+)
+
+// simple struct for registering new entity accessor
+type keyvalue struct {
+	readCalled bool
+	writeCalled bool
+}
+
+// generic read function
+func (kv *keyvalue) Read(req *restful.Request, v interface{}) error {
+	kv.readCalled = true
+	return nil
+}
+
+// generic write function
+func (kv *keyvalue) Write(resp *restful.Response, status int, v interface{}) error {
+	t := reflect.TypeOf(v)
+	rv := reflect.ValueOf(v)
+	for ix := 0; ix < t.NumField(); ix++ {
+		sf := t.Field(ix)
+		io.WriteString(resp, sf.Name)
+		io.WriteString(resp, "=")
+		io.WriteString(resp, fmt.Sprintf("%v\n", rv.Field(ix).Interface()))
+	}
+	kv.writeCalled = true
+	return nil
+}
+
+func main() {
+	kv := new(keyvalue)
+
+	// create new entity accessor for application/test+v1 since it is not standard
+	restful.RegisterEntityAccessor("application/test+v1", kv)
```

Comment:
I see my error, thank you for the explanation. I'm currently playing with the code and I've run into an issue with this:
`ws.Route(ws.GET("/version-test").To(helloV1).           
        Consumes("application/json;v=1").   
        Produces("application/json;v=1")) `
When testing with postman, I receive a "406: Not Acceptable" error due to the content-type not matching.
`restful.RegisterEntityAccessor("application/json;v=1", restful.NewEntityAccessorJSON("application/json;v=1"))`
This line causes the content-type to become "application/json" for the Accept header "application/json;v=1". Removing the "Produces" function from the route fixes this issue, but I am unsure if that is the correct way of handling it.

Also, mapping the same path with different Accept headers will force all Accept headers to be treated as the 1st route mapped to that path. So, in the example that @vinayakhulawale provided, a GET with "application/json;v=2" will trigger helloV1, not helloV2. How should this be handled?